### PR TITLE
(MODULES-4225) add strings to ntp module

### PR DESCRIPTION
**Pull request by jbondpdx**
Added strings to init.pp

Made a few very minor changes to the README.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -1,3 +1,61 @@
+# ntp
+#
+# Main class, includes all other classes.
+#
+# @example Declaring the class
+#   include example
```
Comment (HAIL9000) [inline manifests/init.pp]: I think this may have been copy pasted over by mistake?

File: manifests/init.pp [hunk 2]
```diff
@@ -1,3 +1,61 @@
...
+# @param disable_monitor [Boolean] Disables the monitoring facility in NTP. Default value: true.
+# @param driftfile [Stdlib::Absolutepath] Specifies an NTP driftfile. Default value: '/var/lib/ntp/drift' (except on AIX and Solaris).
+# @param fudge [Optional. Array[String]]. Provides additional information for individual clock drivers. Default value: [ ]
+# @param iburst_enable [Boolean] Specifies whether to enable the iburst option for every NTP peer. Default value: false (true on AIX and Debian).
+# @param interfaces [Array[String]]. Specifies one or more network interfaces for NTP to listen on. Default value: [ ].
+# @param interfaces_ignore [Array[String]] Specifies one or more ignore pattern for the NTP listener configuration (for example: all, wildcard, ipv6). Default value: [ ].
+# @param keys [Array[String]] Distributes keys to keys file. Default value: [ ].
+# @param keys_controlkey [Optional. Ntp::Key_id] Specifies the key identifier to use with the ntpq utility. Value in the range of 1 to 65,534 inclusive. Default value: ' '.
+# @ param keys_enable [Boolean] Whether to enable key-based authentication. Default value: false.
```
Comment (HAIL9000) [inline manifests/init.pp]: Typo here, because of the space it's not picking up the comment

File: manifests/init.pp [hunk 3]
```diff
@@ -1,3 +1,61 @@
...
+# @param tos [Boolean] Whether to enable tos options. Default value: false.
+# @param tos_minclock [Optional Integer[1]] Specifies the minclock tos option. Default value: 3.
+# @param tos_minsane [Optional Integer[1]] Specifies the minsane tos option. Default value: 1.
+# @param tos_floor [Optional Integer[1]] Specifies the floor tos option. Default value: 1.
+# @param tos_ceiling [Optional Integer[1]] Specifies the ceiling tos option. Default value: 15.
+# @param tos_cohort [Variant Boolean, Integer[0,1]] Specifies the cohort tos option. Valid options: 0 or 1. Default value: 0.
+# @param tinker [Boolean] Whether to enable tinker options. Default value: false.
+# @parudlc_stratum [Optional Integer[1,15]]. Specifies the stratum the server should operate at when using the undisciplined local clock as the time source. This value should be set to no less than 10 if ntpd might be accessible outside your immediate, controlled network. Default value: 10.am udlc [Boolean] Specifies whether to configure NTP to use the undisciplined local clock as a time source. Default value: false.
```
Comment (HAIL9000) [inline manifests/init.pp]: Typo here too it looks like

**Comment by HAIL9000** [comment]
So when I ran this through puppet strings everything looked great except that four parameters are not documented:

* keys_enable (Boolean)
* udlc (Boolean)
* udlc_stratum (Optional[Integer[1,15]])
* authprov (Optional[String])

`keys_enable` and `udlc_stratum` are captured by the typos noted above. `udlc` maybe just got missed? I see a comment for it in the README. But `authprov` I don't even see in the README... is that just something we don't document?

**Comment by jbondpdx** [comment]
Yeah, I probably missed udlc... authprov, I don't know about. I suspect it's a victim of our manual readme process. What does it do? I can add it.

**Comment by HAIL9000** [comment]
So looks like this originally came from #306, according to the pull request:

> This parameter is used in some versions of NTPd (such as Novell DSfW) to
enable compatibility with W32Time.

**Comment by jbondpdx** [comment]
OK, @HAIL9000 , PR is updated!

**Comment by HAIL9000** [comment]
So the [@api tag](http://www.rubydoc.info/gems/yard/file/docs/Tags.md#api) is not currently supported for classes (it is for functions though). I filed a ticket to include it for classes ([PDOC-156](https://tickets.puppetlabs.com/browse/PDOC-156)), but we could potentially include it in the meantime in addition to the comments I suggested above. That way, when class support is added it will Just Work.

The nice thing about the @api tag is that when it does work it adds this clear warning to users on the generated HTML page:

screen shot 2017-01-11 at 12 26 02 pm

**Comment by jbondpdx** [comment]
@HAIL9000 this is updated again

File: manifests/config.pp [hunk 4]
```diff
@@ -1,4 +1,6 @@
 # Private Class
```
Comment (HAIL9000) [inline manifests/config.pp]: This comment is redundant now I think we can scrap it

File: manifests/service.pp [hunk 5]
```diff
@@ -1,3 +1,5 @@
+# @api private 
+# This class handles the ntp service. Avoid modifying private classes.
 #
```
Comment (HAIL9000) [inline manifests/service.pp]: Absurdly small nitpick, I think we should remove this blank comment line so our formatting is consistent across all the classes

**Pull request merged**